qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -41,27 +40,29 @@
 
 class OpenTelemetryTest(tu.TestResultCollector):
     def setUp(self):
-        while True:
-            with open("trace_collector.log", "rt") as f:
-                data = f.read()
-                if data.count("resource_spans") != EXPECTED_NUM_SPANS:
-                    time.sleep(5)
-                    continue
-                else:
-                    break
-
-        data = data.split("\n")
-        full_spans = [
-            entry.split("POST")[0] for entry in data if "resource_spans" in entry
-        ]
-        self.spans = []
-        self.resource_attributes = []
-        for span in full_spans:
-            span = json.loads(span)
-            self.spans.append(span["resource_spans"][0]["scope_spans"][0]["spans"][0])
-            self.resource_attributes.append(
-                span["resource_spans"][0]["resource"]["attributes"]
+        with open("trace_collector.log", "rt") as f:
+            data = f.read()
+            json_string = re.sub("\n\t{\n\t", "{", data)
+            json_string = re.sub(
+                "resources     : \n\t", "resources     : {\n\t", json_string
+            )
+            json_string = re.sub(
+                "\n  instr-lib     :", "}\n  instr-lib     :", json_string
             )
+            json_string = re.sub(": \n\t", ':"",', json_string)
+            json_string = re.sub(": \n", ':"",', json_string)
+            json_string = re.sub("\n|\n\t", ",", json_string)
+            json_string = re.sub("\t", "", json_string)
+            json_string = re.sub(r"\b([\w.-]+)\b", r'"\1"', json_string)
+            json_string = re.sub('"span" "kind"', '"span kind"', json_string)
+            json_string = re.sub("{,", "{", json_string)
+            json_string = re.sub(",}", "}", json_string)
+            json_string = re.sub("}{", "},{", json_string)
+            json_string = re.sub(
+                '"events"        : {', '"events"        : [{', json_string
+            )
+            json_string = re.sub('},  "links"', '}],  "links"', json_string)
+            self.spans = eval(json_string[:-1])
```

Comment:
Use of `eval` is generally a code smell and opens up for [ACE](https://en.wikipedia.org/wiki/Arbitrary_code_execution). I know this is just a test so it's probably not very important, but maybe there's a better way to achieve this? Happy to chat offline to understand what you're trying to do here.

Reply:
I'll reconsider